File: MovieBookingApp/src/main/java/com/moviebooking/entity/Movie.java
package com.moviebooking.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "movies")
public class Movie {

    @Transient
    public static final String SEQUENCE_NAME="movie_sequence";
    @Id
    private MovieId movieId;
    private int id;
    private double costOfTicket;
    private int noOfTicketsAllotted;
    private int noOfTicketsSold;
    private String ticketStatus="Book ASAP";

}

File: MovieBookingApp_Frontend/BookYourShow/src/app/model/movie.ts
import movieId from "./movieId";

export class Movie {

    movieId: movieId;
    id:number=0;
    imageUrl: string ='';
    costOfTicket: number=0;
    noOfTicketsAllotted: number=0;
    noOfTicketsSold: number=0;
    ticketStatus:string="Book ASAP";
    


    constructor(
        movieId: movieId,
        costOfTicket: number,
        noOfTicketsAllotted: number,
        imageUrl:string,
    ) {
        this.movieId = movieId;
        this.costOfTicket = costOfTicket;
        this.noOfTicketsAllotted = noOfTicketsAllotted;
        this.imageUrl=imageUrl;
    }
}